scripts/process_licenses.py

```diff
@@ -0,0 +1,156 @@
+    dependency_without_license_count = 0
+    dependency_ignored_count = 0
+
+    missing_licenses_str = ""
+
+    target_dir.mkdir(parents=True, exist_ok=True)
+
+    with open(dependency_list_path, "r") as dependency_file_handle:
+        for line in dependency_file_handle.readlines():
+            line = line.strip()
+            if line == "" or line == "The following files have been resolved:":
+                continue
+            dependency_count += 1
+
+            # Line is a string like: "commons-codec:commons-codec:jar:1.15:compile -- module org.apache.commons.codec [auto]"
+            artifact_details = line.split()[0]
+            group_id, artifact_id, _, version, scope = artifact_details.split(":")
+            current_jar = Path(dependency_jars_path, f"{artifact_id}-{version}.jar")
+            if not current_jar.exists() and current_jar.is_file():
+                raise Exception(f"Expected JAR file does not exist: {current_jar}")
+            current_jar_as_zip = ZipFile(current_jar)
+
+            dependency_lookup_key = f"{group_id}:{artifact_id}"
+            if dependency_lookup_key in IGNORED_DEPENDENCIES:
+                dependency_ignored_count += 1
+                continue
+
+            license_found = False
+            for zip_info in current_jar_as_zip.infolist():
+                if zip_info.is_dir():
+                    continue
+                if zip_info.filename in ("META-INF/LICENSE.txt", "META-INF/LICENSE", "META-INF/LICENSE.md"):
+                    license_found = True
+                    dependency_with_license_count += 1
+                    # Extract license to the target directory
+                    zip_info.filename = f"LICENSE_{group_id}__{artifact_id}"
+                    current_jar_as_zip.extract(zip_info, target_dir)
+                    break
+                if "license" in zip_info.filename.lower():  # Log potential license matches
+                    print(f"Potential license match: {current_jar} {zip_info}")
+
+            if not license_found:
+                print(f"License not found {current_jar}; using value from ADDITIONAL_LICENSES_MAP")
+                license_name = ADDITIONAL_LICENSES_MAP.get(dependency_lookup_key)
+                if license_name:
+                    dependency_without_license_count += 1
+                    missing_licenses_str += f"{dependency_lookup_key}: {license_name}\n"
+                else:
+                    raise Exception(f"The dependency {dependency_lookup_key} does not ship a license file, but neither is it not defined in ADDITIONAL_LICENSES_MAP")
+
+    with open(Path(target_dir, "ADDITIONAL_LICENCES"), "w") as additional_licenses_handle:
+        additional_licenses_handle.write(missing_licenses_str)
+
+    if dependency_count < 30:
+        raise Exception(f"Suspiciously low number of dependency JARs detected in {dependency_jars_path}: {dependency_count}")
+    print("License generation finished")
```

Comment:
I dont see this print statements in confluent runs in github actions, is that expected?

Reply:
The logs are present but you have to go into the job logs
https://github.com/snowflakedb/snowflake-kafka-connector/actions/runs/7874696489/job/21484913973 and then open section `Build With Unit Test for Confluent`.